monai/utils/deprecated.py

```diff
@@ -180,6 +179,9 @@ def _decorator(func):
 
         @wraps(func)
         def _wrapper(*args, **kwargs):
+            if new_name is not None and name in kwargs:
+                # replace the deprecated arg "name" with "new_name"
+                kwargs[new_name] = kwargs[name]
```

Comment:
this change is needed for the required args, for example if we rename the required argument `dim` to `sp_dim`:

deprecated API
```py
def func(dim):
``` 
new API
```py
def func(sp_dim, dim=None):
```
if the user provides `dim` in `func`, it still has the issue of `sp_dim` not being provided in the new version.

So, the deprecated decorator now replace the user-specified deprecated arg with the corresponding latest arg